src/SFA.DAS.LevyTransferMatching.Web/Controllers/PledgesController.cs

```diff
@@ -142,14 +143,45 @@ public async Task<IActionResult> Location(LocationRequest request)
         [Route("create/location")]
         public async Task<IActionResult> Location(LocationPostRequest request)
         {
-            Dictionary<int, string> errors = await _orchestrator.ValidateLocations(request);
+            var multipleValidLocations = new Dictionary<int, IEnumerable<string>>();
+
+            var errors = await _orchestrator.ValidateLocations(request, multipleValidLocations);
+
             if (errors.Any())
             {
                 AddLocationErrorsToModelState(errors);
-                return RedirectToAction("Location", new { request.EncodedAccountId, request.AccountId, request.CacheKey });
+
+                return RedirectToAction(nameof(Location), new { request.EncodedAccountId, request.AccountId, request.CacheKey });
+            }
+
+            await _orchestrator.UpdateCacheItem(request);
```

Comment:
I think this leads to a bug whereby clicking the back link can result in invalid (ambiguous) locations being held in the cache item, which could then be saved to the pledge/db.

This cache item should only hold valid data at all times.

I _think_ this could be fixed by moving the update down below the next if{} block.